compiler/semstmts.nim

```diff
@@ -215,7 +215,11 @@ proc semTry(c: PContext, n: PNode; flags: TExprFlags): PNode =
 
       if a.len == 2 and a[0].kind == nkBracket:
         # rewrite ``except [a, b, c]: body`` -> ```except a, b, c: body```
-        a.sons[0..0] = a[0].sons
+        let exceptionBracket = a[0]
+        a.sons.setLen(exceptionBracket.len+1)
+        a[^1] = move(a[1])
+        for i in 0..<exceptionBracket.len:
+          a[i] = exceptionBracket[i]
```

Comment:
The problem lies in b being an openArray
```nim
proc `[]=`*[T; U, V: Ordinal](s: var seq[T], x: HSlice[U, V], b: openArray[T]) =
  ## Slice assignment for sequences.
  ##
  ## If `b.len` is not exactly the number of elements that are referred to
  ## by `x`, a `splice`:idx: is performed.
  runnableExamples:
    var s = @"abcdefgh"
    s[1 .. ^2] = @"xyz"
    assert s == @"axyzh"

  when defined(nimSeqsV2):
    var b = @b
  ......
```
this works too.

Reply:
I don't understand this fix. Please elaborate.

Reply:
`a.sons[0..0] = a[0].sons` calls `[]=` which seems not to work well with ORC. The problem seems to be related with the parameter `b` of openArray types. Namely, if I add `var b = @b` to `[]=` proc, it works. 

Since there is only a problem with the test case below and It is a bit hard to reproduce it with a simple case, I simply fix it in the compiler. 
```nim
doAssert(not compiles(
  try: 
    echo 1
  except [KeyError as ex1, ValueError as ex2]:
    echo 2
))
```

Reply:
I will try another solution later.